lighthouse-core/report/v2/report-renderer.js

```diff
@@ -181,6 +182,36 @@ class ReportRenderer {
   }
 
   /**
+   * @param {string} text
+   * @return {!Element}
+   */
+  _convertMarkdownLinksToElement(text) {
+    const element = this._createElement('span');
+
+    // Split on markdown links (e.g. [some link](https://...)).
+    const parts = text.split(/(\[(.*?)\]\((https?:\/\/.*?)\))/g);
+
+    while (parts.length) {
+      // Remove the same number of elements as there are capture groups.
```

Comment:
I always forget that `splice` affects both creates a new array *and* modifies the original, so maybe change the comment to something like `// Pop off the same number of elements as there are capture groups.` to make it clear?